hydrolib/core/dflowfm/bc/models.py

```diff
@@ -845,7 +845,7 @@ class RealTime(StrEnum):
     """str: Realtime data source, externally provided"""
 
 
-ForcingData = Union[float, RealTime, ForcingModel]
+ForcingData = Union[float, List[float], RealTime, ForcingModel]
```

Comment:
Where do you need a List of floats on a forcingData line?
(May the confusion came from `SourceSink.zsource` (which indeed may be a list of two floats), but that is no forcingData. So, can we get rid of it here?

Reply:
The `discharge`, `temperaturedelta`, `salinitydelta` you suggested to use `ForcingData` class with those instead of the `Union[float, List[float]]`

Reply:
here https://github.com/Deltares/HYDROLIB-core/pull/721#discussion_r1907148543

Reply:
As discussed: only list[float] can still go, because the ForcingModel stores an entire .bc file, which is all we need for timeseries.

Reply:
this is now back to Union[float, List[floats]] and will be changes to use `forcingData` in a separate merge